### PR TITLE
#3 Switching hash_bucket into defaultdict(list).

### PR DESCRIPTION
**Pull request by corona10**
* To avoid check KeyError, Change hash_bucket from dict to defaultdict.

Signed-off-by: Dong-hee Na <donghee.na92@gmail.com>

### CODE REVIEW

**Comment by corona10** [comment]
For reviewer,
I locally checked that unit test were passed except #1.
I think that this CI failure is not related to this PR.

**Comment by pombredanne** [comment]
@corona10 Thanks! let me review this tomorrow and merge this in.

File: src/samecode/halohash.py [hunk 1]
```diff
@@ -296,24 +297,19 @@ def build_buckets(self):
...
-                hash_buckets[hi] = []
-                hash_buckets[hi].append(lo)
+            hash_buckets[hi].append(lo)
         buckets = []
         # TODO: are we off by one on this range?
         for i in range(self.number_of_buckets):
-            try:
-                buck = hash_buckets[i]
+             buck = hash_buckets[i]
+             if len(buck) > 0:
```
Comment (pombredanne) [inline src/samecode/halohash.py]: Is this not changing the semantics? Would there be a way to have a unit test may be for this function?

Comment (pombredanne) [inline src/samecode/halohash.py]: Actually the semantics are the same after a bit of thinking. 
But then `if buck:` would be enough of a check IMHO

Comment (pombredanne) [inline src/samecode/halohash.py]: Bu then even simpler: `buckets.append(buck or None)` would have the same effect

Comment (corona10) [inline src/samecode/halohash.py]: @pombredanne 
1. `test_BaseBucketHaloHash(self)` is one of the unit test in `test_halohash.py`.
2. I think that most of codes in this repo using 4space indentation. but this section of code is using 3space.. which is the standard of your projects?
3. `buckets.append(buck or None)` looks great. I will changed it.

Comment (pombredanne) [inline src/samecode/halohash.py]: we use 4 spaces. PEP8. Good catch!

Comment (corona10) [inline src/samecode/halohash.py]: @pombredanne It looks my mistake, I used 5space. I will fix it.

**Comment by corona10** [comment]
@pombredanne I fixed it. PTAL if you have a time :)

**Comment by pombredanne** [comment]
LTGM! Thanks.  Merging

**Pull request merged**